### PR TITLE
feat(gitlab): add getPullRequest, getPullRequestFiles, getPullRequestFromBranch, createComment, getComment, updateComment and getUser

### PR DESCRIPTION
**Pull request by jaysomani**
## Summary

Implements the remaining pull request, comment, and user methods for the GitLab adapter, completing the full adapter implementation.

### New methods implemented
- `getPullRequest` — fetches a merge request by iid, normalized to match Gitea/GitHub shape
- `getPullRequestFiles` — fetches changed files using `/diffs` endpoint with `patch_id_sha` polling to handle GitLab's async diff processing
- `getPullRequestFromBranch` — finds an open merge request by source branch
- `createComment` — creates a note on a merge request via GitLab notes API
- `getComment` — retrieves a note by ID, searching across all MRs in the repository
- `updateComment` — updates an existing note by ID
- `getUser` — fetches a user by username via GitLab users search API

### Tests added
- Full test coverage for all new methods including invalid/edge case scenarios
- GitLab-specific tests: `testGetEventPushMatchesCheckoutSha`, `testValidateWebhookEventUsesPlainToken`, `testCreateOrganization`
- Parity tests matching Gitea: `testGetUser`, `testGetUserWithInvalidUsername`, `testCreatePrivateRepository`, `testGetRepositoryWithNonExistingOwner`, `testCreateRepositoryWithInvalidName`, `testDeleteRepositoryTwiceFails`, `testDeleteNonExistingRepositoryFails`, `testGetPullRequestFromBranchNoPR`, `testCreateCommentInvalidPR`, `testGetCommentInvalidId`

## Notes

- GitLab notes (comments) are scoped to merge requests — `getComment` and `updateComment` search across all MRs to find the note by ID since the MR iid is not stored with the comment ID
- `getPullRequestFiles` uses `patch_id_sha` polling per GitLab's official documentation recommendation for async diff processing
- `getUser` returns `username` field (not `login` like Gitea/GitHub) — this is a GitLab API difference
- This PR is based on `feat/gitlab-adapter-webhooks` and will need a rebase once that PR merges

### CODE REVIEW

File: src/VCS/Adapter/Git/GitLab.php [hunk 1]
```diff
@@ -414,32 +414,144 @@ public function createBranch(string $owner, string $repositoryName, string $newB
...
     {
-        throw new Exception("Not implemented");
+        $ownerPath = $this->getOwnerPath($owner);
+        $projectPath = urlencode("{$ownerPath}/{$repositoryName}");
+        $apiUrl = "/projects/{$projectPath}/hooks";
+    
+        $payload = [
+            'url' => $url,
+            'token' => $secret,
+            'enable_ssl_verification' => false,
```
[the remark this answers is not in the document]
Comment (jaysomani) [reply src/VCS/Adapter/Git/GitLab.php:448]: Consistent with the Gitea adapter which also does not enforce SSL verification. This is intentional for the self-hosted test environment. SSL verification can be addressed as a follow-up improvement across all adapters.

File: src/VCS/Adapter/Git/GitLab.php [hunk 2]
```diff
@@ -470,24 +600,130 @@ public function getOwnerName(string $installationId, ?int $repositoryId = null):
...
+            throw new Exception("Failed to get merge request: HTTP {$statusCode}");
+        }
+
+        $mr = $response['body'] ?? [];
+
+        // Normalize to match expected shape (consistent with Gitea/GitHub)
+        return [
+            'number'  => $mr['iid'] ?? 0,
+            'title'   => $mr['title'] ?? '',
+            'state'   => $mr['state'] ?? '',
```
[the remark this answers is not in the document]
Comment (jaysomani) [reply src/VCS/Adapter/Git/GitLab.php:595]: The Gitea adapter also returns the raw API state value directly without normalization. Normalizing state values consistently across all adapters (GitHub, Gitea, GitLab) would be a separate refactor PR to avoid introducing partial normalization. GitLab uses 'opened' while Gitea uses 'open' — this difference is documented and callers should handle both values.

**Review by Meldiron — APPROVED** [review]
